jupyterhub/app.py

```diff
@@ -2371,6 +2442,26 @@ async def init_role_assignment(self):
                     # explicitly defined list
                     # ensure membership list is exact match (adds and revokes permissions)
                     setattr(role, kind, orm_role_bearers)
+                    # if the role_spec was contributed by the authenticator, mark the newly
+                    # created assignments as managed by authenticator too; also mark the
+                    # assignment as not stale (in case if it was marked as such initially)
+                    if role_spec.get('managed_by_auth', False):
+                        entity_name = kind[:-1]
+                        association_class = orm._role_associations[entity_name]
+                        kind_id = getattr(association_class, f'{entity_name}_id')
+                        associations = (
+                            db.query(association_class)
+                            .filter(
+                                kind_id.in_([bearer.id for bearer in orm_role_bearers])
+                                & (association_class.role_id == role.id)
+                            )
+                            .all()
```

Comment:
I htink you can drop this `.all()`, since the `for` will iterate over the rows. Right?

Reply:
Done in 9dc91fb707a846cef9106b3ce6dfeed08d650246.